src/app/shares/viewshareaccount/viewshareaccount.component.ts

```diff
@@ -0,0 +1,354 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+
+/** Routing Imports */
+import { ActivatedRoute, Router } from '@angular/router';
+
+/** Custom Imports */
+import { SharesService } from '../shares.service';
+
+/** Custom Models */
+import { PurchasedShare } from './purchased-shares.model';
+import { Charge } from './charges.model';
+import { Dividend } from './dividends.model';
+
+/** Common model for similar tables */
+export interface CommonTable {
+  details: string;
+  value: string;
+}
+
+/**
+ * View Share Account Component
+ */
+@Component({
+  selector: 'mifosx-viewshareaccount',
+  templateUrl: './viewshareaccount.component.html',
+  styleUrls: ['./viewshareaccount.component.scss']
+})
+export class ViewshareaccountComponent implements OnInit {
+
+  shareAccountId: any;
+  shareAccountDetails: any;
+  date: any = {};
+  staffData: any = {};
+  status: any;
+  choice: any;
+  dividends: any;
+  showDividends: any;
+  dividendsTableDataSource: Dividend[] = [];
+  chargeAction: any;
+  chargePayAction: any;
+  sharesPendingForApproval: any;
+  purchasedSharesTableShow: any;
+  purchasedSharesTableDataSource: PurchasedShare[] = [];
+  purchasedShares: any;
+  charges: any;
+  chargeTableShow: any;
+  chargeTableDataSource: Charge[] = [];
+  accountTableDataSource: CommonTable[] = [];
+  approveSharesTableDataSource: CommonTable[] = [];
+  buttons: any = {};
+  annualChargeId: any;
+  displayedColumns: any = {
+    purchasedSharesTable: ['transactionDate', 'transactionType', 'totalShares', 'purchasedOrRedeemedPrice', 'chargeAmount', 'amountRecievedOrReturned'],
+    chargeTable: ['name', 'feeOrPenalty', 'paymentDueAt', 'calculationType', 'due', 'paid', 'waived', 'outstanding'],
+    dividendsTable: ['transactionDate', 'amount', 'transactionRef', 'status'],
+    accountTable: ['details', 'values'],
+    approvedSharesTable: ['details', 'values']
+  };
+
+  /**
+   * Retrieves the Share Account data.
+   * Sets Button according to the status of account.
+   * Adds data to the tables.
+   * @param {Router} router Router.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {SharesService} SharesService Loan Service.
+   */
+  constructor(private router: Router,
+              private route: ActivatedRoute,
+              private sharesService: SharesService) {
+    this.route.data.subscribe((shareAccountData: {
+      shareAccount: any
+    }) => {
+      const data = shareAccountData.shareAccount;
+      this.shareAccountDetails = data;
+      this.convertDateArrayToObject('date');
+      this.staffData.staffId = data.staffId;
+      this.date.toDate = new Date();
+      this.date.fromDate = new Date(data.timeline.activatedDate);
+      this.status = data.status.value;
+
```

Comment:
There is too much happening in the constructor. Please add comments to explain. Maybe shift into proper functions and call them from ngOnInit?